tests/bluetooth/audio/cap_commander/src/test_broadcast_reception.c

```diff
@@ -61,16 +89,31 @@ static void *cap_commander_test_broadcast_reception_setup(void)
 
 static void cap_commander_test_broadcast_reception_before(void *f)
 {
+	int err;
+	struct cap_commander_test_broadcast_reception_fixture *fixture = f;
+
 	memset(f, 0, sizeof(struct cap_commander_test_broadcast_reception_fixture));
-	cap_commander_test_broadcast_reception_fixture_init(f);
+	cap_commander_test_broadcast_reception_fixture_init(fixture);
+
+	for (size_t i = 0; i < ARRAY_SIZE(fixture->conns); i++) {
+		err = bt_cap_commander_discover(&fixture->conns[i]);
+		zassert_equal(0, err, "Unexpected return value %d", err);
+	}
 }
 
 static void cap_commander_test_broadcast_reception_after(void *f)
 {
 	struct cap_commander_test_broadcast_reception_fixture *fixture = f;
+	int err;
 
 	bt_cap_commander_unregister_cb(&mock_cap_commander_cb);
+	bt_bap_broadcast_assistant_unregister_cb(&fixture->broadcast_assistant_cb);
 
+	/* We need to cleanup since the CAP commander remembers state */
+	if (bt_cap_common_proc_is_active()) {
+		err = bt_cap_commander_cancel();
+		zassert_equal(0, err, "Unexpected return value: %d", err);
+	}
```

Comment:
```suggestion
	bt_cap_commander_cancel();
```

We can just always attempt to cancel and ignore any errors (or ignore `-EALREADY`). We don't need to hack `bt_cap_common_proc_is_active` to check whether it's active or not.